modules/phase_field/src/ics/MultiSmoothSuperellipsoidIC.C

```diff
@@ -0,0 +1,168 @@
+// each semiaxis can be varied by a uniform or normal distribution
+
+
+#include "MultiSmoothSuperellipsoidIC.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<MultiSmoothSuperellipsoidIC>()
+{
+  InputParameters params = validParams<SmoothSuperellipsoidBaseIC>();
+  params.addClassDescription("Random distribution of smooth ellipse with given minimum spacing");
+  params.addRequiredParam<unsigned int>("numbub", "The number of bubbles to place");
+  params.addRequiredParam<Real>("bubspac", "minimum spacing of bubbles, measured from center to center");
+  params.addParam<unsigned int>("numtries", 1000, "The number of tries");
+  params.addRequiredParam<Real>("semiaxis_a", "Mean semiaxis value in the x direction for the ellipse");
+  params.addRequiredParam<Real>("semiaxis_b", "Mean semiaxis value in the y direction for the ellipse");
+  params.addParam<Real>("exponent", 2,"Exponent n for each superellipsoid, n=2 is a normal ellipse");
+  params.addParam<Real>("semiaxis_c", 1, "Mean semiaxis value in the z direction for the ellipse");
+  params.addParam<Real>("semiaxis_a_variation", 0.0, "Plus or minus fraction of random variation in the bubble semiaxis in the x direction for uniform, standard deviation for normal");
+  params.addParam<Real>("semiaxis_b_variation", 0.0, "Plus or minus fraction of random variation in the bubble semiaxis in the y direction for uniform, standard deviation for normal");
+  params.addParam<Real>("semiaxis_c_variation", 0.0, "Plus or minus fraction of random variation in the bubble semiaxis in the z direction for uniform, standard deviation for normal");
+  MooseEnum rand_options("uniform normal none","none");
+  params.addParam<MooseEnum>("semiaxis_variation_type", rand_options, "Type of distribution that random superellipsoid semiaxes will follow");
+  return params;
+}
+
+MultiSmoothSuperellipsoidIC::MultiSmoothSuperellipsoidIC(const InputParameters & parameters) :
+    SmoothSuperellipsoidBaseIC(parameters),
+    _numbub(getParam<unsigned int>("numbub")),
+    _bubspac(getParam<Real>("bubspac")),
+    _numtries(getParam<unsigned int>("numtries")),
+    _exponent(getParam<Real>("exponent")),
+    _semiaxis_a(getParam<Real>("semiaxis_a")),
+    _semiaxis_b(getParam<Real>("semiaxis_b")),
+    _semiaxis_c(getParam<Real>("semiaxis_c")),
+    _semiaxis_a_variation(getParam<Real>("semiaxis_a_variation")),
+    _semiaxis_b_variation(getParam<Real>("semiaxis_b_variation")),
+    _semiaxis_c_variation(getParam<Real>("semiaxis_c_variation")),
+    _semiaxis_variation_type(getParam<MooseEnum>("semiaxis_variation_type"))
+{
+}
+
+void
+MultiSmoothSuperellipsoidIC::initialSetup()
+{
+
+  //Set up domain bounds with mesh tools
+  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+  {
+    _bottom_left(i) = _mesh.getMinInDimension(i);
+    _top_right(i) = _mesh.getMaxInDimension(i);
+  }
+  _range = _top_right - _bottom_left;
+
+  switch (_semiaxis_variation_type)
+  {
+  case 2: //No variation
```

Comment:
Thanks for drawing my attention to that piece of code. I fixed it to longer use a switch, also i noticed that as it was written the breaks would have prevented it from working as intended anyway.